docs/datasets/data-anonymization/index.md

```diff
@@ -0,0 +1,277 @@
+
+blur:
+  kernel_size: 31 # Kernel size for the Gaussian blur (int)
+  sigma_x: 11 # Sigma x for the Gaussian blur (int)
+```
+
+- `config/yolo_create_dataset.yaml`
+
+```yaml
+rosbag:
+  input_bags_folder: "/path/to/input_bag_folder" # Path to the input ROS 2 bag files folder
+
+dataset:
+  output_dataset_folder: "/path/to/output/dataset" # Path to the output dataset folder
+  output_dataset_subsample_coefficient: 25 # Subsample coefficient for the dataset (int)
+
+grounding_dino:
+  box_threshold: 0.1 # Threshold for the bounding box (float)
+  text_threshold: 0.1 # Threshold for the text (float)
+  nms_threshold: 0.1 # Threshold for the non-maximum suppression (float)
+
+open_clip:
+  score_threshold: 0.7 # Validity threshold for the OpenCLIP model (float
+
+bbox_validation:
+  iou_threshold: 0.9 # Threshold for the intersection over union (float), if the intersection over union is greater than this threshold, the object will be selected as inside the validation prompt
+```
+
+- `config/yolo_train.yaml`
+
+```yaml
+dataset:
+  input_dataset_yaml: "path/to/data.yaml" # Path to the config file of the dataset, which is created in the previous step
+
+yolo:
+  epochs: 100 # Number of epochs for the YOLOv8 model (int)
+  model: "yolov8x.pt" # Select the base model for YOLOv8 ('yolov8x.pt' 'yolov8l.pt', 'yolov8m.pt', 'yolov8n.pt')
+```
+
+- `config/yolo_anonymize.yaml`
+
+```yaml
+rosbag:
+  input_bag_path: "/path/to/input_bag/bag.mcap" # Path to the input ROS 2 bag file with 'mcap' or 'sqlite3' extension
+  output_bag_path: "/path/to/output_bag_file" # Path to the output ROS 2 bag folder
+  output_save_compressed_image: True # Save images as compressed images (True or False)
+  output_storage_id: "sqlite3" # Storage id for the output bag file (`sqlite3` or `mcap`)
+
+yolo:
+  model: "path/to/yolo/model" # Path to the trained YOLOv8 model file (`.pt` extension) (you can download the pre-trained model from releases)
+  config_path: "path/to/input/data.yaml" # Path to the config file of the dataset, which is created in the previous step
+  confidence: 0.15 # Confidence threshold for the YOLOv8 model (float)
+
+blur:
+  kernel_size: 31 # Kernel size for the Gaussian blur (int)
+  sigma_x: 11 # Sigma x for the Gaussian blur (int)
+```
+
+## Usage
+
+The tool provides two options to anonymize images in ROS 2 bag files.
+
+### Option 1: Anonymize with Unified Model
+
+You should provide a single rosbag and tool anonymize images in rosbag with a unified model. The model is a combination
+of GroundingDINO, OpenCLIP, YOLOv8 and SegmentAnything. If you don't want to use pre-trained YOLOv8 model, you can
+follow the instructions in the second option to train your own YOLOv8 model.
+
+You should set your configuration in config/anonymize_with_unified_model.yaml file.
+
+```bash
+python3 main.py config/anonymize_with_unified_model.yaml --anonymize_with_unified_model
```

Comment:
When I tried running this command, I got the following error message:

```
torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 1024.00 MiB. GPU 0 has a total capacity of 10.87 GiB of which 1010.88 MiB is free. Including non-PyTorch memory, this process has 8.66 GiB memory in use. Of the allocated memory 8.21 GiB is allocated by PyTorch, and 266.44 MiB is reserved by PyTorch but unallocated. If reserved but unallocated memory is large try setting PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True to avoid fragmentation.  See documentation for Memory Management  (https://pytorch.org/docs/stable/notes/cuda.html#environment-variables)
```

The error disappeared after setting the environment variable as suggested (`export PYTORCH_CUDA_ALLOC_CONF=expandable_segments:True`). If this is expected, it might be user friendly to add in the instruction to set the variable first before running the command.

Reply:
I knew this error was caused by the GPU running out of memory, but I didn't know it could be resolved this way. Thank you. I’ve added it under a new 'Troubleshooting' section.

Reply:
btw, 'Troubleshooting' section add both package `readme.md` and documentation page.